components/org.wso2.carbon.identity.api.server.branding.preference.management/org.wso2.carbon.identity.api.server.branding.preference.management.v1/src/main/java/org/wso2/carbon/identity/api/server/branding/preference/management/v1/core/BrandingPreferenceManagementService.java

```diff
@@ -0,0 +1,396 @@
+            throw handleConfigurationMgtException(e, ERROR_CODE_ERROR_GETTING_BRANDING_PREFERENCE, resourceName);
+        } catch (IOException e) {
+            throw handleException
+                    (Response.Status.INTERNAL_SERVER_ERROR, ERROR_CODE_ERROR_BUILDING_RESPONSE_EXCEPTION, null);
+        }
+    }
+
+    /**
+     * Update branding preferences.
+     *
+     * @param brandingPreferenceModel Branding Preference Model with new preferences.
+     */
+    public void updateBrandingPreference(BrandingPreferenceModel brandingPreferenceModel) {
+
+        String tenantDomain = getTenantDomainFromContext();
+        // Check whether the branding resource exists in the particular tenant.
+        if (!isResourceExists(BRANDING_RESOURCE_TYPE, getDefaultResourceName())) {
+            throw handleException(Response.Status.NOT_FOUND, ERROR_CODE_BRANDING_PREFERENCE_NOT_EXISTS, tenantDomain);
+        }
+
+        String preferencesJson = generatePreferencesJsonFromRequest(brandingPreferenceModel.getPreference());
+        if (!BrandingPreferenceUtils.isValidJsonString(preferencesJson)) {
+            throw handleException(Response.Status.BAD_REQUEST, ERROR_CODE_INVALID_BRANDING_PREFERENCE, null);
+        }
+
+        try {
+            InputStream inputStream = BrandingPreferenceUtils.generatePreferenceInputStream(preferencesJson);
+            Resource brandingPreferenceResource =
+                    buildResourceFromBrandingPreference(brandingPreferenceModel, inputStream);
+            BrandingPreferenceServiceHolder.getBrandingPreferenceConfigManager()
+                    .replaceResource(BRANDING_RESOURCE_TYPE, brandingPreferenceResource);
+        } catch (ConfigurationManagementException e) {
+            throw handleConfigurationMgtException(e, ERROR_CODE_ERROR_UPDATING_BRANDING_PREFERENCE, tenantDomain);
+        } catch (JsonProcessingException e) {
+            throw handleException(Response.Status.INTERNAL_SERVER_ERROR, ERROR_CODE_JSON_PROCESSING_EXCEPTION,
+                    e.getMessage());
+        } catch (UnsupportedEncodingException e) {
+            throw handleException(Response.Status.INTERNAL_SERVER_ERROR, ERROR_CODE_UNSUPPORTED_ENCODING_EXCEPTION,
+                    e.getMessage());
+        }
+    }
+
+    /**
+     * Check whether a branding preference resource already exists with the same name in the particular tenant.
+     *
+     * @param resourceType Resource type.
+     * @param resourceName Resource name.
+     * @return Return true if the resource already exists. If not return false.
+     */
+    private boolean isResourceExists(String resourceType, String resourceName) {
+
+        Resource resource;
+        try {
+            resource = BrandingPreferenceServiceHolder.getBrandingPreferenceConfigManager()
+                    .getResource(resourceType, resourceName);
+        } catch (ConfigurationManagementException e) {
+            return false;
```

Comment:
Why don't we throw the error or handle here?

Reply:
If config manager throws a resource not found error, return false only for the specific error code.

Reply:
Addressed with https://github.com/wso2/identity-api-server/pull/332/commits/95afc9eceedc9e6e0e0e8eec8bb23634936f37c1